extensions/amp-analytics/0.1/events.js

```diff
@@ -1384,15 +1391,44 @@ export class VideoEventTracker extends EventTracker {
         event.target,
         'No target specified by video session event.'
       );
-      targetReady.then((target) => {
-        if (!target.contains(el)) {
-          return;
-        }
-        const normalizedDetails = removeInternalVars(details);
-        listener(new AnalyticsEvent(target, normalizedType, normalizedDetails));
+
+      Promise.all(targetReadyPromises).then((targetReady) => {
+        targetReady.forEach((target) => {
+          if (!target[0].contains(el)) {
+            return;
+          }
+          const normalizedDetails = removeInternalVars(details);
+          listener(
+            new AnalyticsEvent(target[0], normalizedType, normalizedDetails)
+          );
+        });
       });
     });
   }
+
+  /**
+   * Assert that the selectors are all unique and are non-empty
+   * @param {!Array<string>|string} selectors
+   */
+  assertValidSelectors_(selectors) {
+    if (isArray(selectors)) {
+      if (selectors.length === 0) {
+        userAssert(
+          selectors.length,
+          'Missing required selector on video trigger'
+        );
+      }
```

Comment:
Besides the error handling, is there anything different in this method from the other already existing `assertUniqueSelectors_`?

If not, then we should just refactor the other `assertUniqueSelectors_` to have empty array error handling, and have a generic error message, and then use that method instead (to avoid dupe code)

Reply:
great. just made the change.